controllers/integrationpipeline/integrationpipeline_adapter_test.go

```diff
@@ -970,4 +970,124 @@ var _ = Describe("Pipeline Adapter", Ordered, func() {
 			Expect(intgPipelineRunWithDeletionTimestamp.DeletionTimestamp).ToNot(BeNil())
 		})
 	})
+
+	When("GetIntegrationPipelineRunStatus is called with an Integration PLR with invalid TEST_OUTPUT result", func() {
+		var (
+			taskRunInvalidResult      *tektonv1.TaskRun
+			intgPipelineInvalidResult *tektonv1.PipelineRun
+		)
+
+		BeforeEach(func() {
+
+			taskRunInvalidResult = &tektonv1.TaskRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-taskrun-invalid",
+					Namespace: "default",
+				},
+				Spec: tektonv1.TaskRunSpec{
+					TaskRef: &tektonv1.TaskRef{
+						Name: "test-taskrun-invalid",
+						ResolverRef: tektonv1.ResolverRef{
+							Resolver: "bundle",
+							Params: tektonv1.Params{
+								{
+									Name:  "bundle",
+									Value: tektonv1.ParamValue{Type: "string", StringVal: "quay.io/redhat-appstudio/example-tekton-bundle:test"},
+								},
+								{
+									Name:  "name",
+									Value: tektonv1.ParamValue{Type: "string", StringVal: "test-task"},
+								},
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, taskRunInvalidResult)).Should(Succeed())
+
+			now := time.Now()
+			taskRunInvalidResult.Status = tektonv1.TaskRunStatus{
+				TaskRunStatusFields: tektonv1.TaskRunStatusFields{
+					StartTime:      &metav1.Time{Time: now},
+					CompletionTime: &metav1.Time{Time: now.Add(5 * time.Minute)},
+					Results: []tektonv1.TaskRunResult{
+						{
+							Name: "TEST_OUTPUT",
+							Value: *tektonv1.NewStructuredValues(`{
+												"result": "INVALID",
+											}`),
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Status().Update(ctx, taskRunInvalidResult)).Should(Succeed())
+
+			intgPipelineInvalidResult = &tektonv1.PipelineRun{
```

Comment:
An idea, not a blocker for review: maybe we don't need to create a new pipelineRun for this. We could just use an existing one and change its status to point to the task with the invalid result, especially as `GetIntegrationPipelineRunStatus` doesn't require the PLR to be present on the cluster.

Reply:
I tried, but something is interfering with it.

When I run make test:
```
Pipeline Adapter when GetIntegrationPipelineRunStatus is called with an Integration PLR with invalid TEST_OUTPUT result [It] ensures test status in snapshot is updated to failed
/home/mbasti/upstream-repos/integration-service/controllers/integrationpipeline/integrationpipeline_adapter_test.go:1053

  [FAILED] Expected
      <integrationteststatus.IntegrationTestStatus>: 7  # TestSuccess
  to equal
      <integrationteststatus.IntegrationTestStatus>: 6
  In [It] at: /home/mbasti/upstream-repos/integration-service/controllers/integrationpipeline/integrationpipeline_adapter_test.go:1057 
```

However, when I do step by step in debugger, test succeeded with fail.

So I'd keep new PLR here for easier life :D

Reply:
Yup, fair enough.

Reply:
We'd probably need to deepcopy the PLR to make it work, so not much gained aside from lines of code.

Reply:
I tried deepcopy too, got 1 passing run, 1 the same test failure as mentioned previously